flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlotTable.java

```diff
@@ -462,7 +471,7 @@ public int freeSlot(AllocationID allocationId, Throwable cause) throws SlotNotFo
 	 * @return True if the timeout is valid; otherwise false
 	 */
 	public boolean isValidTimeout(AllocationID allocationId, UUID ticket) {
-		checkInit();
+		checkStartedOrClosing();
```

Comment:
Same here, this might allow a race condition between this method and the stopping of the `TimerService`.

Reply:
I think this would not hurt if we relax this check to `STARTED` (not `CREATED`) and make `RUNNING` state check part of the return value. This method is called after `notifyTimeout` where we can also relax it to `STARTED`. Even if we schedule `timerService#stop` to the main thread, `notifyTimeout` can be called afterwards because `timerService#stop` does not block on internal executor shutdown and we do not want it, I guess. If those methods are called afterwards, it should not be problem as they will just find that there is no registered callback anymore.